packages/expo-camera/ios/EXCamera/Utilities/EXCameraUtils.m

```diff
@@ -121,19 +135,41 @@ + (NSString *)writeImage:(NSData *)image toPath:(NSString *)path
   return [fileURL absoluteString];
 }
 
-+ (void)updateImageSampleMetadata:(CMSampleBufferRef)imageSampleBuffer withAdditionalData:(NSDictionary *)additionalData inResponse:(NSMutableDictionary *)response
++ (NSData *)writeImageToData:(UIImage *)image withExifData:(NSDictionary *)exif imageQuality:(float)quality
 {
-  CFDictionaryRef exifAttachments = CMGetAttachment(imageSampleBuffer, kCGImagePropertyExifDictionary, NULL);
-  NSDictionary *metadata = (__bridge NSDictionary *)exifAttachments;
-  [self updateExifMetadata:metadata withAdditionalData:additionalData inResponse:response];
+  // Get metadata (includes the EXIF data)
+  CGImageSourceRef sourceCGIImageRef = CGImageSourceCreateWithData((CFDataRef) UIImageJPEGRepresentation(image, 1.0f), NULL);
+  NSDictionary *sourceMetadata = (__bridge NSDictionary *) CGImageSourceCopyPropertiesAtIndex(sourceCGIImageRef, 0, NULL);
+  
+  NSMutableDictionary *updatedMetadata = [sourceMetadata mutableCopy];
+  
+  // Add camera exif data
+  [updatedMetadata setObject:exif forKey:(__bridge NSString *)kCGImagePropertyExifDictionary];
+
+  // Set compression quality
+  [updatedMetadata setObject:@(quality) forKey:(__bridge NSString *)kCGImageDestinationLossyCompressionQuality];
+
+  // Create an image destination
+  NSMutableData *processedImageData = [NSMutableData data];
+  CGImageDestinationRef destinationCGImageRef = CGImageDestinationCreateWithData((__bridge CFMutableDataRef)processedImageData, CGImageSourceGetType(sourceCGIImageRef), 1, NULL);
+
+  // Add image to the destination
+  // Note: it'll save only these value which are under the kCGImagePropertyExif* key.
+  CGImageDestinationAddImage(destinationCGImageRef, image.CGImage, (__bridge CFDictionaryRef) updatedMetadata);
```

Comment:
ANd what about `kCGImagePropertyGPSDictionary`?

Reply:
@lukmccall, well you've resolved this thread, but I still don't know what about `kCGImagePropertyGPSDictionary`?

Reply:
`kCGImagePropertyGPSDictionary` will be saved too. I've changed logic a little bit. Now we save into image file every piece of information which are under kCGImageProperty* key.